test/unit/spec/narrative_core/kbaseNarrativeDataList-spec.js

```diff
@@ -4,11 +4,177 @@
+                var objData = [];
+
+                // mock service calls
+                mockServiceWizard();
+                mockNarrativeServiceListObjects(objData);
+                mockWorkSpaceService();
+
+                // Create datalist object
+                dataListObj = new DataList($dataList, {});
+                expect(dataListObj).toBeDefined();
+
+                // Populate datalist object via the refresh function
+                dataListObj.ws_name = 'some_workspace';
+                dataListObj.ws = new Workspace(Config.url('workspace'), null);
+                await dataListObj.refresh();
+            });
+
+            it('Should instantiate itself', () => {
+                expect(dataListObj).toBeDefined();
+                expect($dataList.html()).toContain('This Narrative has no data yet.');
+            });
+
+
+            it('Should render the add data text button ', () => {
+                var $addDataButton = $dataList.find('.kb-data-list-add-data-text-button');
+                expect($addDataButton).toBeDefined();
+                expect($addDataButton.html()).toContain('Add Data');
+            });
+        });
+
+        describe('With data', () => {
+            beforeEach( async () => {
+                var objData = [
+                    {
+                        'object_info': [
+                            5,
+                            'Rhodobacter_CACIA_14H1',
+                            'KBaseGenomes.Genome-7.0',
+                            '2020-10-03T01:15:14+0000',
+                            1,
+                            'testuser',
+                            54640,
+                            'testuser:narrative_1601675739009',
+                            '53af8071b814a1db43f81eb490a35491',
+                            3110399,
+                            {}
+                        ]
+                    }
+                ];
+
+                // mock service calls
+                mockServiceWizard();
+                mockNarrativeServiceListObjects(objData);
+                mockWorkSpaceService();
+
+                // Create datalist object
+                const dataListObj = new DataList($dataList, {});
+
+                // Populate datalist object via the refresh function
+                dataListObj.ws_name = 'some_workspace';
+                dataListObj.ws = new Workspace(Config.url('workspace'), null);
+                await dataListObj.refresh();
+            });
+
+            it('Should instantiate itself', () => {
+                expect(dataListObj).toBeDefined();
+            });
+
+            it('Should render a data object row', () => {
+                expect($dataList.find('.narrative-card-row')).toBeDefined();
+            });
+
+            it('Should generate the add data button ', () => {
+                var $addDataButton = $dataList.find('.kb-data-list-add-data-button');
+                expect($addDataButton).toBeDefined();
```

Comment:
is it possible to check the colour of the button?

Reply:
It is possible :) Generally I try and test features/functionality in unit tests though. For example, if the color changed based on the state of the page, then it would be important to test the color changed at the appropriate time. I decided not to test this color because it is static.

If you would like me to test the color though, I definitely can :)

Reply:
It might be best to verify that there's a button, there's only one, and it's the color we want.
You could do that with:
```
expect($addDataButton.length).toEqual(1);
expect($addDataButton.is('button')).toBeTruthy();
```

For color checking, I'm not sure what's the best answer. It probably involves unpacking the DOM element from the jquery node and using `window.getComputedStyle`, like `getComputedStyle($addDataButton[0]).backgroundColor`.

Reply:
You can also access the css property values by using the jquery function `.css()` i.e. `element.css( "backgroundColor" )`

Reply:
I still think I would like to hold off on testing the color of the button since the color is static. Testing static colors can more comprehensively be tested via snapshot tests if that is something we would like to look into. I think unit tests are more aimed at testing functionality and the implementation.

Reply:
I don't know that I am that sold on checking the colour of the button, because it'll make it a nuisance to change the tests if the UI colour scheme changes, but it might be good to add a test to check that the button changes colour when certain events occur (if the button is supposed to change colour) -- i.e. make sure that `element.css("backgroundColor")` before !== `element.css("backgroundColor")` after `event`.

Reply:
Yeah, agreed, that's better as a snapshot test.